ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/metadata.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.plugin_utils.strip_empties import strip_empties_from_list
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+from .utils import UtilsMixin
+
+
+class MetadataMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def metadata(self) -> dict | None:
+        """
+        Generate metadata.cv_pathfinder.internet_exit_policies if available.
+        """
+        if not self._filtered_internet_exit_policies:
+            return None
+
+        internet_exit_polices = []
+        for internet_exit_policy in self._filtered_internet_exit_policies:
+            # Currently only supporting zscaler
+            if internet_exit_policy["type"] != "zscaler":
+                continue
+
+            ufqdn, ipsec_key = self._get_ipsec_credentials(internet_exit_policy)
+            internet_exit_polices.append(
+                {
+                    "name": internet_exit_policy["name"],
+                    "type": internet_exit_policy["type"],
+                    "city": get(
+                        self.shared_utils.wan_site,
+                        "resolved_location.city",
+                        required=True,
+                        org_key="cv_pathfinder_regions[].sites[].resolved_location.city",
+                    ),
+                    "country": get(
+                        self.shared_utils.wan_site,
+                        "resolved_location.country",
+                        required=True,
+                        org_key="cv_pathfinder_regions[].sites[].resolved_location.country",
+                    ),
+                    "upload_bandwidth": get(internet_exit_policy, "zscaler.upload_bandwidth"),
+                    "download_bandwidth": get(internet_exit_policy, "zscaler.download_bandwidth"),
+                    "firewall": get(internet_exit_policy, "zscaler.firewall.enabled", default=False),
+                    "ips_control": get(internet_exit_policy, "zscaler.firewall.ips", default=False),
```

Comment:
We have defaults here but not in the schema

Reply:
Right, I made the schema as lightweight as possible, but I can add some more details to it.

Reply:
added required fields. The defaults here are on the output, not input, so should not really be in the schema.